cmd/embedded-cluster/join.go

```diff
@@ -368,3 +393,100 @@ func runK0sInstallCommand(fullcmd string) error {
 	}
 	return nil
 }
+
+// maybeEnableHA checks if the cluster has more than 3 nodes and prompts the user
+// to enable high availability if it is not already enabled.
+func maybeEnableHA(ctx context.Context) error {
+	kcli, err := kubeutils.KubeClient()
+	if err != nil {
+		return fmt.Errorf("unable to create kube client: %w", err)
+	}
+	embeddedclusterv1beta1.AddToScheme(kcli.Scheme())
+	isHA, err := isHAEnabled(ctx, kcli)
+	if err != nil {
+		return fmt.Errorf("unable to check if HA is enabled: %w", err)
+	}
+	if isHA {
+		return nil
+	}
+	hostname, err := os.Hostname()
+	if err != nil {
+		return fmt.Errorf("unable to get hostname: %w", err)
+	}
+	numNodes, err := waitForNode(ctx, kcli, hostname)
```

Comment:
This should probably be outside of this function with a loading message like: "Waiting for node to join the cluster" or something.

Reply:
Do you think we should apply this _and_ the `waitForK0s` to all joins instead of it being wrapped in the `enable-ha` condition?

Reply:
Yes